### PR TITLE
Plane label annotator

### PR DESCRIPTION
**Pull request by kevinyamauchi**
This PR adds an annotator for painting labels on a rendering plane.

Based on: https://github.com/napari/napari/pull/4346 by @alisterburt 

https://github.com/napari-threedee/napari-threedee/assets/1120672/5cd902aa-df30-426a-92c7-3ecae70c9027

### CODE REVIEW

File: src/napari_threedee/annotators/label/annotator.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        labels_layer: napari.layers.Labels
+    ):
+        self.image_layer = image_layer
+        self.labels_layer = labels_layer
+
+    def _on_mode_change(self, event):
+        if event.mode in ("paint", "erase"):
+            # replace the mouse callback with the one aware of the image layer rendering plane
+            remove_mouse_callback_safe(self.labels_layer.mouse_drag_callbacks, napari_draw)
+            add_mouse_callback_safe(self.labels_layer.mouse_drag_callbacks, self.draw)
```
Comment (alisterburt) [inline src/napari_threedee/annotators/label/annotator.py:48]: 🐒

File: src/napari_threedee/annotators/label/annotator.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        self.labels_layer.events.mode.connect(self._on_mode_change)
+
+        # update the event callbacks
+        if self.labels_layer.mode in ("paint", "erase"):
+            # replace the mouse callback with the one aware of the image layer rendering plane
+            remove_mouse_callback_safe(self.labels_layer.mouse_drag_callbacks, napari_draw)
+            add_mouse_callback_safe(self.labels_layer.mouse_drag_callbacks, self.draw)
+
+        # currently the plane painting must be done with n_edit_dims = 3
+        # todo modify painting so it can be done 2D in plane
```
Comment (alisterburt) [inline src/napari_threedee/annotators/label/annotator.py:65]: 👍🏼

File: src/napari_threedee/annotators/label/annotator.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            start, end = layer.get_ray_intersections(
+                position=event.position,
+                view_direction=event.view_direction,
+                dims_displayed=layer._dims_displayed,
+                world=True,
+            )
+            if start is None and end is None:
+                return None
+            coordinates = first_nonzero_coordinate(layer.data, start, end)
+        return coordinates
```
Comment (alisterburt) [inline src/napari_threedee/annotators/label/annotator.py:130]: logic here is super clean, nice work

**Review by alisterburt — COMMENTED** [review]
dude! I'm psyched on this, great job

Looks good to me, will merge as tests are passing ❤️ 

edit: too slow, already merged 😆

File: src/napari_threedee/annotators/label/annotator.py (hunk 3 above)
Comment (kevinyamauchi) [reply src/napari_threedee/annotators/label/annotator.py:130]: Haha, I'm glad you think so, @alisterburt  - it's largely copied from your PR to napari 😆

**Comment by kevinyamauchi** [comment]
> edit: too slow, already merged 😆

haha, sorry, I got too excited. yolo!

**Pull request merged**